### ISSUE TITLE
Update logits in-place during guided generation instead of masking

### ISSUE THREAD

**Issue by brandonwillard** [outlines-dev/outlines#317]
We should consider updating the logit arrays in-place instead of creating separate mask arrays and adding those to the logits.
